docs/testing/webdriverio/02-unit-testing.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Unit Testing
+sidebar_label: Unit Testing
+description: Unit Testing
+slug: /testing/webdriverio/unit-testing
+---
+
+# Unit Testing
+
+WebdriverIO makes it easy to unit test components and app utility functions in the browser. Unit tests validate the code in isolation. Well written tests are fast, repeatable, and easy to reason about. It tries to follow a simple guiding principle: the more your tests resemble the way your software is used, the more confidence they can give you.
+
+### Test Setup
+
+To resemble how your component is being used as close as possible to reality we need to render it into an actual DOM tree. WebdriverIO provides a helper package for this that you can use called `@wdio/browser-runner/stencil`. It exports a `render ` method that allows us to mount our component to the DOM.
+
+For example, given the following component:
+
+```ts reference title="/src/components/my-component/my-component.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.tsx
+```
+
+We import the component into our test to render it in the browser:
+
+```ts reference title="/src/components/my-component/my-component.test.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.test.tsx#L2-L18
+```
+
+You can find more information about the `render` methods option and its return object in the WebdriverIO [documentation](https://webdriver.io/docs/component-testing/stencil#render-options).
```

Comment:
Between these docs linked here and this page, it's not clear if I need to register child components in `render`'s `components` field. If I have a `my-component` that renders another component `my-child`, would I need to register `my-child` as well here if I were testing `my-component`?

Reply:
That's a good point. I highlighted that it provides flexibility to leave out components that have side effects and can cause flakiness:

<img width="755" alt="Screenshot 2024-02-08 at 11 17 08 AM" src="https://github.com/ionic-team/stencil-site/assets/731337/a4db875a-615a-48d4-bf3b-86d02e4edbb0">